cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesExport.java

```diff
@@ -0,0 +1,408 @@
+        writer.writeEndElement();
+    }
+
+    private static void writePowerFlows(Network network, XMLStreamWriter writer) {
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getLoadStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getGeneratorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getShuntCompensatorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getStaticVarCompensatorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getBatteryStream);
+
+        network.getDanglingLineStream().forEach(dl -> dl.getAliasFromType(CGMES_PREFIX_ALIAS + "Terminal_Boundary")
+                .ifPresent(terminal -> writePowerFlow(terminal, dl.getP0(), dl.getQ0(), writer)));
+        // TODO what about flows of dl's generations?
+    }
+
+    private static <I extends Injection<I>> void writeInjectionPowerFlowToCgmes(Network network, XMLStreamWriter writer, Function<Network, Stream<I>> getInjectionStream) {
+        getInjectionStream.apply(network).forEach(i -> writePowerFlow(i.getTerminal(), writer));
+    }
+
+    private static void writePowerFlow(Terminal terminal, XMLStreamWriter writer) {
+        String cgmesTerminal = ((Connectable<?>) terminal.getConnectable()).getAliasFromType(CGMES_PREFIX_ALIAS + CgmesNames.TERMINAL1).orElse(null);
+        if (cgmesTerminal != null) {
+            writePowerFlow(cgmesTerminal, terminal.getP(), terminal.getQ(), writer);
+        } else if (terminal.getConnectable() instanceof Load && terminal.getConnectable().isFictitious()) {
+            Load svInjection = (Load) terminal.getConnectable();
+            writeSvInjection(svInjection, writer);
+        } else {
+            LOG.error("No defined CGMES terminal for {}", terminal.getConnectable().getId());
+        }
+    }
+
+    private static void writePowerFlow(String terminal, double p, double q, XMLStreamWriter writer) {
+        try {
+            writer.writeStartElement(CIM_NAMESPACE, "SvPowerFlow");
+            writer.writeAttribute(RDF_NAMESPACE, ID, getUniqueId());
+            writer.writeStartElement(CIM_NAMESPACE, "SvPowerFlow.p");
+            writer.writeCharacters(String.valueOf(p));
+            writer.writeEndElement();
+            writer.writeStartElement(CIM_NAMESPACE, "SvPowerFlow.q");
+            writer.writeCharacters(String.valueOf(q));
+            writer.writeEndElement();
+            writer.writeEmptyElement(CIM_NAMESPACE, "SvPowerFlow.Terminal");
+            writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + terminal);
+            writer.writeEndElement();
+        } catch (XMLStreamException e) {
+            throw new UncheckedXmlStreamException(e);
+        }
+    }
+
+    private static void writeSvInjection(Load svInjection, XMLStreamWriter writer) {
+        try {
+            writer.writeStartElement(CIM_NAMESPACE, "SvInjection");
```

Comment:
in CGMES `SvInjection` is reserved for reporting mismatches in buses, not for reporting values of `EquivalentInjections` (of boundary nodes). The value of the equivalent injection at the boundary must be entered as a `SvPowerFlow` value at the corresponding `Terminal` of the `EquivalentInjection`. Change will be proposed in a later commit

Reply:
Writing `SvInjections` is ok, as we have converted to fictitious loads only the values that we have seen in the original CGMES input as `SvInjections` also. Sorry for the noise.